src/metabase/driver/ddl/interface.clj

```diff
@@ -0,0 +1,58 @@
+(ns metabase.driver.ddl.interface
+  (:require [clojure.string :as str]
+            [metabase.driver :as driver]
+            [metabase.util.i18n :refer [tru]]))
+
+(defn schema-name
+  "Returns a schema name for persisting models. Needs the database to use the db id and the site-uuid to ensure that
+  multiple connections from multiple metabae remain distinct. The UUID will have the first character of each section taken.
+
+  (schema-name {:id 234} \"143dd8ce-e116-4c7f-8d6d-32e99eaefbbc\") ->  \"metabase_cache_1e483_1\""
+  [{:keys [id] :as _database} uuid-string]
+  (let [instance-string (apply str (map first (str/split uuid-string #"-")))]
+    (format "metabase_cache_%s_%s" instance-string id)))
+
+(defmulti format-name
+  "Transform a lowercase string Table or Field name in a way appropriate for this dataset (e.g., `h2` would want to
+  upcase these names; `mongo` would want to use `\"_id\"` in place of `\"id\"`. This method should return a string.
+  Defaults to an identity implementation."
+  {:arglists '([driver table-or-field-name])}
+  driver/dispatch-on-initialized-driver
+  :hierarchy #'driver/hierarchy)
```

Comment:
This looks like it was just moved over from the test utils. I don't think this docstring makes sense. **Why** would H2 _need_ to upcase names? Why does Mongo need to convert `id` to `_id`? 

Is the purposes of this for escaping disallowed characters? Can we use something existing like `driver/escape-alias` or `driver/escape-entity-name-for-metadata`?

Reply:
Yes this was copied over and the docstring left without any particular check if it was still valid.